src/Service.Tests/Configuration/TelemetryTests.cs

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using static Azure.DataApiBuilder.Service.Tests.Configuration.ConfigurationTests;
+
+namespace Azure.DataApiBuilder.Service.Tests.Configuration;
+
+/// <summary>
+/// Contains tests for telemetry functionality.
+/// </summary>
+[TestClass, TestCategory(TestCategory.MSSQL)]
+public class TelemetryTests
+{
+    public TestContext TestContext { get; set; }
+    private const string TEST_APP_INSIGHTS_CONN_STRING = "InstrumentationKey=testKey;IngestionEndpoint=https://unitTest.com/;LiveEndpoint=https://unittest2.com/";
```

Comment:
nit: unittest.com resolves to a domain. Any way to just specify local host for these endpoints?
also for the InstrumentationKey, isn't usage of that deprecated? is it even needed here?

Reply:
connection string is a combination of all those key/value. direct usage of instrumentation key is deprecated.

Reply:
done,thanks